crates/torii/indexer/src/engine.rs

```diff
@@ -207,6 +208,9 @@ pub struct ParallelizedEvent {
     pub event: Event,
 }
 
+type TaskPriority = usize;
```

Comment:
Wdyt about an enum, using the `Ord` trait to keep keys ordered in the `BTreeMap`?
Feeling like enum would be more idiomatic, and remove the magic values like we have in the engine.

Reply:
Something like:
```rust
enum TaskPriority {
    Low,
    Medium,
    High,
}
```